spec/features/admin/admin_can_deactivate_user_spec.rb

```diff
@@ -23,4 +23,46 @@
     expect(user.roles.count).to eq(1)
     expect(user.roles[0][:name]).to eq('blocked_user')
   end
+
+  scenario 'admin can deactivate another admin' do
+    user.roles.create(name: 'admin')
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(admin)
+    question2 = Fabricate(:question, user: admin)
+
+    visit question_path(question)
+
+    click_button "Deactivate User"
+    expect(user.roles.count).to eq(1)
+    expect(user.roles[0][:name]).to eq('blocked_user')
+  end
+
+  scenario 'admin can reactivate a blocked user' do
+    user.roles.create(name: 'blocked_user')
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(admin)
+    question2 = Fabricate(:question, user: admin)
+
+    visit question_path(question)
+
+    click_button "Reactivate User"
+    expect(user.roles.count).to eq(1)
+    expect(user.roles[0][:name]).to eq('registered_user')
+  end
+
+  scenario 'admin cannot deactivate self' do
```

Comment:
Nice work testing the edge cases! This makes sense that an admin can't ban themselves.